lib/iris/fileformats/grib/message.py

```diff
@@ -256,24 +255,21 @@ def __getitem__(self, keys):
                 # Only the non-masked values are included in codedValues.
                 _data = np.empty(shape=bitmap.shape)
                 _data[bitmap.astype(bool)] = data
-                # `np.ma.masked_array` masks where input = 1, the opposite of
-                # the behaviour specified by the GRIB spec.
-                data = ma.masked_array(_data, mask=np.logical_not(bitmap))
+                # Use nan where input = 1, the opposite of the behaviour
+                # specified by the GRIB spec.
+                _data[np.logical_not(bitmap.astype(bool))] = np.nan
```

Comment:
@lbdreyer Okay, so we explicitly set masked points to `nan` here, and maintain a `nan` filled `ndarray`, rather than a masked array ... got it.